source/vibe/web/internal/rest/jsclient.d

```diff
@@ -30,81 +58,98 @@ import std.conv : to;
 
 	auto intf = RestInterface!TImpl(settings, true);
 
-	output.formattedWrite("%s = new function() {\n", name.length ? name : intf.I.stringof);
+	auto fout = indentSink(output, jsgenset.indentStep);
 
-	output.put("var toRestString = function(v) { return v; }\n");
+	fout.formattedWrite("%s%s = new function() {\n", jsgenset.parent ? "" : "this.",
+			jsgenset.name.length ? jsgenset.name : intf.I.stringof);
+
+	if (jsgenset.parent)
+		fout.put("var toRestString = function(v) { return JSON.stringify(v); }\n");
 
 	foreach (i, SI; intf.SubInterfaceTypes) {
-		output.generateInterface!SI(__traits(identifier, intf.SubInterfaceFunctions[i]), intf.subInterfaces[i].settings);
+		fout.put("\n");
+		auto childjsset = jsgenset.child(__traits(identifier, intf.SubInterfaceFunctions[i]));
+		fout.generateInterface!SI(intf.subInterfaces[i].settings, childjsset);
 	}
 
 	foreach (i, F; intf.RouteFunctions) {
 		alias FT = FunctionTypeOf!F;
 		auto route = intf.routes[i];
 
 		// function signature
-		output.formattedWrite("  this.%s = function(", route.functionName);
+		fout.put("\n");
+		fout.formattedWrite("this.%s = function(", route.functionName);
 		foreach (j, param; route.parameters) {
-			output.put(param.name);
-			output.put(", ");
+			fout.put(param.name);
+			fout.put(", ");
 		}
-		static if (!is(ReturnType!FT == void)) output.put("on_result, ");
-		output.put("on_error) {\n");
+		static if (!is(ReturnType!FT == void)) fout.put("on_result, ");
+
+		fout.put("on_error) {\n");
 
 		// url assembly
+		fout.put("var url = ");
 		if (route.pathHasPlaceholders) {
-			output.put("    var url = ");
-			output.serializeToJson(intf.baseURL);
+			fout.serializeToJson(intf.baseURL);
 			foreach (p; route.pathParts) {
-				output.put(" + ");
-				if (!p.isParameter) output.serializeToJson(p.text);
-				else output.formattedWrite("encodeURIComponent(toRestString(%s))", p.text);
+				fout.put(" + ");
+				if (!p.isParameter) fout.serializeToJson(p.text);
+				else fout.formattedWrite("encodeURIComponent(toRestString(%s))", p.text);
 			}
-			output.put(";\n");
 		} else {
-			output.formattedWrite("    var url = %s;\n", Json(concatURL(intf.baseURL, route.pattern)));
+			auto rpn = route.parameters
```

Comment:
That was indeed broken. I've committed a fix for the `hasPathPlaceholders` computation: https://github.com/rejectedsoftware/vibe.d/commit/f23d7740e7e459e6a890cf6d7ef64bb1b5640402
And also one for the JS URL computation: https://github.com/rejectedsoftware/vibe.d/commit/10e37edd880e3790e5cc6ff7ee89e3018348b3ad

Can you remove the changes in the else block from the PR and rebase?